### PR TITLE
[mysql-connector-cpp] Add new port

### PR DESCRIPTION
**Pull request by Xiaobin0860**
**Describe the pull request**

- #### What does your PR fix?
  No, this PR is going to add a new port - [mysql\-connector\-cpp](https://github.com/mysql/mysql-connector-cpp)
  https://github.com/microsoft/vcpkg/issues/17674#issue-876073714
  https://github.com/microsoft/vcpkg/issues/11921#issue-638211577

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  x64-windows,x64-windows-static,x64-windows-static-md,x64-linux,x64-linux-dynamic,x64-osx,x64-osx-dynamic SUPPORTED, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/mysql-connector-cpp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(GLOB DLLS ${CURRENT_PACKAGES_DIR}/debug/lib/*.dll)
+if(DLLS)
+    file(COPY ${DLLS} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin)
+    file(REMOVE ${DLLS})
+endif()
+file(REMOVE ${CURRENT_PACKAGES_DIR}/debug/BUILDINFO ${CURRENT_PACKAGES_DIR}/debug/LICENSE ${CURRENT_PACKAGES_DIR}/debug/README)
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/mysql-connector-cpp RENAME copyright)
```
Comment (Adela0814) [inline ports/mysql-connector-cpp/portfile.cmake:69]:
```suggestion
file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

```

File: ports/mysql-connector-cpp/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "mysql-connector-cpp",
+  "version": "1.1.13",
+  "description": "This is a release of MySQL Connector/C++, the C++ interface for communicating with MySQL servers.",
+  "homepage": "https://github.com/mysql/mysql-connector-cpp",
+  "supports": "!(windows & x86) & !uwp",
+  "dependencies": [
```
Comment (Adela0814) [inline ports/mysql-connector-cpp/vcpkg.json:7]: Please add license for `mysql-connector-cpp`.

File: scripts/ci.baseline.txt [hunk 3]
```diff
@@ -690,6 +690,9 @@ msmpi:x64-uwp=fail
 munit:arm-uwp=fail
 munit:arm64-windows=fail
 munit:x64-uwp=fail
+mysql-connector-cpp:x64-osx=fail
+mysql-connector-cpp:x64-windows-static=fail
+mysql-connector-cpp:x64-windows-static-md=fail
```
Comment (Adela0814) [inline scripts/ci.baseline.txt:695]: Why you add those failed?

File: ports/mysql-connector-cpp/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,69 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mysql/mysql-connector-cpp
+
+    # 8.0.29 ea056780b92c4adfc151a0f3d74f0cf35d454af2a6b43bb67de6be0288f9c2c4ecdd520611fd50d16ee2f43daa1ce86e963adf646b6974e46f73c99452c2eecc
```
Comment (Adela0814) [inline ports/mysql-connector-cpp/portfile.cmake:5]: Please remove this comment.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: scripts/ci.baseline.txt (hunk 3 above)
Comment (Xiaobin0860) [reply scripts/ci.baseline.txt]: windows static compilation errors have been fixed
[the dependence \[icu\] build failure](https://github.com/microsoft/vcpkg/issues/24806#issue-1242037634), osx support has been removed

File: ports/mysql-connector-cpp/portfile.cmake (hunk 1 above)
Comment (Xiaobin0860) [reply ports/mysql-connector-cpp/portfile.cmake]: already apply these changes

**Pull request closed without merging**